packages/chakra-helpers/src/CodeBlockTabs.tsx

```diff
@@ -0,0 +1,177 @@
+import {getIconComponent} from './helpers';
+import {usePrismTheme} from './prism';
+
+function getTabButtonProps(
+  loc: 'LEFT' | 'RIGHT',
+  visible: boolean,
+  gradientColor: TinyColor
+): FlexProps {
+  // const gradientColor = colorMode === 'light' ? '255,255,255' : '0,0,0';
+
+  const endColor = gradientColor.clone().setAlpha(0);
+
+  return {
+    pos: 'absolute',
+    left: loc === 'LEFT' ? '-1px' : undefined,
+    right: loc === 'RIGHT' ? '-1px' : undefined,
+    top: '0',
+    bottom: '0',
+    alignItems: 'center',
+    justifyContent: loc === 'LEFT' ? 'flex-start' : 'flex-end',
+    zIndex: '99',
+    width: '8',
+    background: `linear-gradient(${[
+      loc === 'LEFT' ? '90deg' : '270deg',
+      `${gradientColor.toRgbString()} 50%`,
+      endColor.toRgbString()
+    ]})`,
+    opacity: visible ? 1 : 0,
+    pointerEvents: visible ? 'all' : 'none',
+    transition: 'all 250ms ease-in-out',
+    cursor: 'pointer'
+  };
+}
+
+interface CodeBlockTabsProps {
+  languages: string[];
+  activeLanguage: string;
+}
+
+export const CodeBlockTabs = ({
+  languages,
+  activeLanguage
+}: CodeBlockTabsProps): JSX.Element => {
+  // Track inner (infinite width) and outer (container width) boxes using refs
+  const outerRef = useRef<HTMLDivElement>(null);
+  const innerRef = useRef<HTMLDivElement>(null);
+  // Track the scroll position, represented as [0,1] inclusive
+  const [scrollPosition, setScrollPosition] = useState(0);
+
+  // Track tabs scroll position to determine if arrows are necessary
+  useEffect(() => {
+    if (!outerRef.current) return;
+    const el = outerRef.current;
+    const onScroll = () => {
+      const maxScroll = el.scrollWidth - el.clientWidth;
+      setScrollPosition(el.scrollLeft / maxScroll);
+    };
+    outerRef.current.addEventListener('scroll', onScroll);
+    return () => {
+      el.removeEventListener('scroll', onScroll);
+    };
+  }, [outerRef]);
+
+  // Allow for arrow presses to induce a short scroll left or right
+  const bumpScroll = (distance: number) => () => {
+    if (!outerRef) return;
+    outerRef.current.scrollBy({
+      left: distance,
+      behavior: 'smooth'
+    });
+  };
+
+  const {setLanguage} = useContext(MultiCodeBlockContext);
+
+  const theme = usePrismTheme();
+  const bgColor = useToken('colors', 'gray.800');
+  const gradientColor = useColorModeValue(
+    new TinyColor('white'),
+    new TinyColor(bgColor)
+  );
+
+  // Determine which arrows (if any) need to be shown
+  let showArrows = false;
+  if (
+    innerRef.current &&
+    outerRef.current &&
+    innerRef.current.clientWidth > outerRef.current.clientWidth
+  ) {
+    showArrows = true;
+  }
+  // Determine if the left or right arrows should be shown based on overall
+  // visibility, and the scroll position
+  const showLeftArrow = showArrows && scrollPosition > 0;
+  const showRightArrow = showArrows && scrollPosition < 1;
+
+  return (
+    <Box pos="relative" pt="1">
```

Comment:
Since the arrow buttons within this element have a `zIndex`, it's important that we "zero-out" the z-index on the parent element so that the z-index'd elements don't overlap other elements on the page (like the header).

```suggestion
    <Box pos="relative" pt="1" zIndex="0">
```